tests/test.hpp

```diff
@@ -154,6 +156,39 @@ struct dummy_fixture {};
 
 #define STR(text) PUGIXML_TEXT(text)
 
+#ifdef PUGIXML_CHAR8_MODE
+#	if defined(__clang__) || defined(__GNUC__)
+#		define ALIASING_BARRIER(ptr) asm volatile("" : : "rm"(ptr) : "memory")
+#	else
+#		define ALIASING_BARRIER(ptr)
+#	endif
+inline const char8_t* char_cast(const char* bytes)
+{
+	ALIASING_BARRIER(bytes);
```

Comment:
Is this addressing a specific problem? It's really hard to tell without any comments.